src/journal/test/JournalTest.hs

```diff
@@ -44,36 +45,79 @@ startJournalFake = FakeJournal Vector.empty 0
 
 appendBSFake :: ByteString -> FakeJournal -> (FakeJournal, Either AppendError ())
 appendBSFake bs fj@(FakeJournal bss ix)
-  | unreadBytes < limit = (FakeJournal (Vector.snoc bss bs) ix, Right ())
+  | position < limit =
+    trace (unlines [ "TRACE"
+                   , "ix: " ++ show ix
+                   , "readBytes: " ++ show readBytes
+                   , "unreadBytes: " ++  show unreadBytes
+                   , "readPadBytes: " ++  show readPadBytes
+                   , "unreadPadBytes: " ++ show unreadPadBytes
+                   , "position: " ++ show position
+                   , "limit: " ++ show limit
+                   ])
+    (FakeJournal (Vector.snoc bss bs) ix, Right ())
   | otherwise           = (fj, Left BackPressure)
   where
     termLen :: Int
     termLen = oTermBufferLength testOptions
 
+    position = readBytes + unreadBytes
+
     limit :: Int
-    limit = termLen `div` 2
+    limit = readBytes + termLen `div` 2
+
+    readBytes :: Int
+    readBytes = Vector.sum
+              . Vector.cons readPadBytes
```

Comment:
This is test code so it it fine but just so that you are aware, `Vector.cons` is an O(n) operation (since it need to copy the tail). But as I said, it should be fine in test code.

Reply:
Thanks, fixed!